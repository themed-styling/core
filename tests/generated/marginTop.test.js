import 'jest-styled-components'

import { marginTop } from '../../src/index.js'

import { testAllOn } from '../utilities.js'

testAllOn(marginTop, 'marginTop')
