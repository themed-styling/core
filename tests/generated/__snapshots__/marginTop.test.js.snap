// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marginTop renders 0 1`] = `
Array [
  .c0 {
  margin-top: 0;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: 0;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: 0;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: 0;
}

<div
    className="c0"
  />,
]
`;

exports[`marginTop renders function 1`] = `
Array [
  .c0 {
  margin-top: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: has;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: has;
}

<div
    className="c0"
  />,
]
`;

exports[`marginTop renders invalid theme path 1`] = `
Array [
  .c0 {
  margin-top: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: invalid.path.1;
}

<div
    className="c0"
  />,
]
`;

exports[`marginTop renders negative number 1`] = `
Array [
  .c0 {
  margin-top: -54881px;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: -54881px;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: -42365px;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: -42365px;
}

<div
    className="c0"
  />,
]
`;

exports[`marginTop renders object [breakpoints] 1`] = `
Array [
  @media screen and (min-width:November) {
  .c0 {
    margin-top: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    margin-top: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    margin-top: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    margin-top: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {
  .c0 {
    margin-top: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    margin-top: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    margin-top: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    margin-top: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    margin-top: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    margin-top: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    margin-top: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    margin-top: November;
  }
}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    margin-top: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    margin-top: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    margin-top: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    margin-top: November;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`marginTop renders object [breakpoints] 2`] = `
Array [
  @media May {
  .c0 {
    margin-top: July;
  }
}

@media April {
  .c0 {
    margin-top: August;
  }
}

@media January {
  .c0 {
    margin-top: September;
  }
}

@media April {
  .c0 {
    margin-top: November;
  }
}

@media June {
  .c0 {
    margin-top: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {
  .c0 {
    margin-top: July;
  }
}

@media April {
  .c0 {
    margin-top: August;
  }
}

@media January {
  .c0 {
    margin-top: September;
  }
}

@media April {
  .c0 {
    margin-top: November;
  }
}

@media June {
  .c0 {
    margin-top: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    margin-top: January;
  }
}

@media April {
  .c0 {
    margin-top: December;
  }
}

@media January {
  .c0 {
    margin-top: April;
  }
}

@media April {
  .c0 {
    margin-top: May;
  }
}

@media June {
  .c0 {
    margin-top: June;
  }
}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    margin-top: January;
  }
}

@media April {
  .c0 {
    margin-top: December;
  }
}

@media January {
  .c0 {
    margin-top: April;
  }
}

@media April {
  .c0 {
    margin-top: May;
  }
}

@media June {
  .c0 {
    margin-top: June;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`marginTop renders postitive number 1`] = `
Array [
  .c0 {
  margin-top: 54881px;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: 54881px;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: 42365px;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: 42365px;
}

<div
    className="c0"
  />,
]
`;

exports[`marginTop renders string 1`] = `
Array [
  .c0 {
  margin-top: July;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: July;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: June;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: June;
}

<div
    className="c0"
  />,
]
`;

exports[`marginTop renders valid theme path 1`] = `
Array [
  .c0 {
  margin-top: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  margin-top: DarkSlateBlue;
}

<div
    className="c0"
  />,
]
`;
